### PR TITLE
Openamp xlnx assist update oct29 2025

### PR DESCRIPTION
**Pull request by bentheredonethat**
@zeddii  please wait on merge

@tnmysh  please take a look

this PR

1. preserve backward compatibility
2. enable new driver -- tested in build and boot with latest zynqmp BSP and openamp demo on target 

thanks

### CODE REVIEW

**Comment by zeddii** [comment]
Waiting! Will hold for @tnmysh to comment.

File: lopper/assists/openamp_xlnx.py [hunk 1]
```diff
@@ -1074,12 +1076,161 @@ def xlnx_remoteproc_construct_carveouts(tree, carveouts, new_ddr_nodes, verbose
...
+    power_domains = []
+    cluster_ranges_val = []
+
+    cluster_reg = 0xff9a0000
+    driver_compat_str  = {
+      SOC_TYPE.ZYNQMP : "xlnx,zynqmp-r5fss",
+      SOC_TYPE.VERSAL : "xlnx,versal-r5fss",
+      SOC_TYPE.VERSAL_NET : "xlnx,versal-net-r52fss",
+    }
+    core_names = { SOC_TYPE.VERSAL_NET: "r52f", SOC_TYPE.VERSAL: "r5f", SOC_TYPE.ZYNQMP: "rf5" }
```
Comment (tnmysh) [inline lopper/assists/openamp_xlnx.py:1117]: %s/rf5/r5f/r

**Comment by tnmysh** [comment]
@bentheredonethat commit message needs to be fixed.

It's not "upstream driver", but "upstream bindings". Please use bindings instead of driver where possible.

File: lopper/assists/openamp_xlnx.py [hunk 2]
```diff
@@ -1074,12 +1076,161 @@ def xlnx_remoteproc_construct_carveouts(tree, carveouts, new_ddr_nodes, verbose
...
+    core_compat_substr = "tcm-1.0"
+    absolute_addr_tcm_substr = "tcm-global-1.0"
+    core_reg_val = []
+    core_reg_names = []
+    core_reg_names_base_str = "tcm0"
+    core_reg_names_index = 0
+    power_domains = []
+    cluster_ranges_val = []
+
+    cluster_reg = 0xff9a0000
```
Comment (tnmysh) [inline lopper/assists/openamp_xlnx.py:1111]: cluster reg isn't in upstream binding I think.

**Comment by bentheredonethat** [comment]
@tnmysh  updated.

File: lopper/assists/openamp_xlnx.py (hunk 2 above)
Comment (bentheredonethat) [reply lopper/assists/openamp_xlnx.py]: done.

File: lopper/assists/openamp_xlnx.py (hunk 1 above)
Comment (bentheredonethat) [reply lopper/assists/openamp_xlnx.py]: done.

**Pull request merged**